recipe/cmake-configure-alignment.patch

```diff
@@ -0,0 +1,95 @@
+     foreach(_substitution_variable ${SEARCHED_VARIABLES})
+         string(TOUPPER ${_substitution_variable} _substitution_variable_upper)
+         if (NOT ${_substitution_variable_upper})
+-            string(REGEX MATCH "m4_define\\(${_substitution_variable}, \\[(.*)\\]" MACTHED_STRING ${configure_line})
++            string(REGEX MATCH "m4_define\\(${_substitution_variable}, *\\[(.*)\\]" MACTHED_STRING ${configure_line})
+             if (CMAKE_MATCH_1)
+                 set(${_substitution_variable_upper} ${CMAKE_MATCH_1})
+             endif()
+@@ -413,6 +413,24 @@
+     endforeach()
+ endforeach()
+ 
++set(PACKAGE_VERSION "${PCRE2_MAJOR}.${PCRE2_MINOR}")
++
++string(REPLACE ":" ";" LIBPCRE2_8_VERSION_LIST ${LIBPCRE2_8_VERSION})
++list(GET LIBPCRE2_8_VERSION_LIST 0 LIBPCRE2_8_VERSION_CURRENT)
++list(GET LIBPCRE2_8_VERSION_LIST 1 LIBPCRE2_8_VERSION_REVISION)
++list(GET LIBPCRE2_8_VERSION_LIST 2 LIBPCRE2_8_VERSION_AGE)
++math(EXPR LIBPCRE2_8_SOVERSION "${LIBPCRE2_8_VERSION_CURRENT} - ${LIBPCRE2_8_VERSION_AGE}")
++math(EXPR LIBPCRE2_8_MACHO_VERSION "${LIBPCRE2_8_VERSION_CURRENT} + 1")
++set(LIBPCRE2_8_VERSION "${LIBPCRE2_8_SOVERSION}.${LIBPCRE2_8_VERSION_AGE}.${LIBPCRE2_8_VERSION_REVISION}")
++
++string(REPLACE ":" ";" LIBPCRE2_POSIX_VERSION_LIST ${LIBPCRE2_POSIX_VERSION})
++list(GET LIBPCRE2_POSIX_VERSION_LIST 0 LIBPCRE2_POSIX_VERSION_CURRENT)
++list(GET LIBPCRE2_POSIX_VERSION_LIST 1 LIBPCRE2_POSIX_VERSION_REVISION)
++list(GET LIBPCRE2_POSIX_VERSION_LIST 2 LIBPCRE2_POSIX_VERSION_AGE)
++math(EXPR LIBPCRE2_POSIX_SOVERSION "${LIBPCRE2_POSIX_VERSION_CURRENT} - ${LIBPCRE2_POSIX_VERSION_AGE}")
++math(EXPR LIBPCRE2_POSIX_MACHO_VERSION "${LIBPCRE2_POSIX_VERSION_CURRENT} + 1")
++set(LIBPCRE2_POSIX_VERSION "${LIBPCRE2_POSIX_SOVERSION}.${LIBPCRE2_POSIX_VERSION_AGE}.${LIBPCRE2_POSIX_VERSION_REVISION}")
++
+ CONFIGURE_FILE(src/pcre2.h.in
+                ${PROJECT_BINARY_DIR}/pcre2.h
+                @ONLY)
+@@ -540,16 +558,36 @@
+ # 8-bit library
+ 
+ IF(PCRE2_BUILD_PCRE2_8)
+-ADD_LIBRARY(pcre2-8 ${PCRE2_HEADERS} ${PCRE2_SOURCES} ${PROJECT_BINARY_DIR}/config.h)
+-SET_PROPERTY(TARGET pcre2-8
+-  PROPERTY COMPILE_DEFINITIONS PCRE2_CODE_UNIT_WIDTH=8)
++  ADD_LIBRARY(pcre2-8 ${PCRE2_HEADERS} ${PCRE2_SOURCES} ${PROJECT_BINARY_DIR}/config.h)
++
++SET_TARGET_PROPERTIES(pcre2-8 PROPERTIES
++  COMPILE_DEFINITIONS PCRE2_CODE_UNIT_WIDTH=8
++  MACHO_COMPATIBILITY_VERSION ${LIBPCRE2_8_MACHO_VERSION}
++  MACHO_CURRENT_VERSION ${LIBPCRE2_8_MACHO_VERSION}
++  VERSION ${LIBPCRE2_8_VERSION}
++  SOVERSION ${LIBPCRE2_8_SOVERSION})
+ SET(targets ${targets} pcre2-8)
+ ADD_LIBRARY(pcre2-posix ${PCRE2POSIX_HEADERS} ${PCRE2POSIX_SOURCES})
+-SET_PROPERTY(TARGET pcre2-posix
+-  PROPERTY COMPILE_DEFINITIONS PCRE2_CODE_UNIT_WIDTH=8)
++SET_TARGET_PROPERTIES(pcre2-posix PROPERTIES
++  COMPILE_DEFINITIONS PCRE2_CODE_UNIT_WIDTH=8
++  MACHO_COMPATIBILITY_VERSION ${LIBPCRE2_POSIX_MACHO_VERSION}
++  MACHO_CURRENT_VERSION ${LIBPCRE2_POSIX_MACHO_VERSION}
```

Comment:
AFAIR, `compatibility_version = current + 1` and `current_version = (current + 1).revision`

Reply:
Checked on master of `pcre2` that this formula holds, only references I found only are other PRs fixing this. Not even the autotools mythbuster seems to mention this.